save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/HistoryView.kt

```diff
@@ -166,6 +263,71 @@ class HistoryView : RComponent<HistoryProps, State>() {
         }
     }
 
+    private fun deleteExecutions() {
+        setState {
+            confirmationType = ConfirmationType.DELETE_CONFIRM
+            isConfirmWindowOpen = true
+            confirmLabel = ""
+            confirmMessage = "Are you sure you want to delete all executions?"
+        }
+    }
+
+    private fun deleteExecutionsBuilder() {
+        val headers = Headers().also {
+            it.set("Accept", "application/json")
+            it.set("Content-Type", "application/json")
+        }
+        GlobalScope.launch {
+            responseFromDeleteExecutions =
+                    post("${window.location.origin}/deleteAllExecution?name=${props.name}&owner=${props.owner}", headers, undefined)
+        }.invokeOnCompletion {
+            if (responseFromDeleteExecutions.ok) {
+                window.location.href = "${window.location.origin}#/${props.owner}/${props.name}"
+            } else {
+                responseFromDeleteExecutions.text().then {
+                    setState {
+                        errorLabel = "Failed to delete executions"
+                        errorMessage = it
+                        isErrorOpen = true
+                    }
+                }
+            }
+        }
+    }
+
+    private fun deleteExecution(id: Long) {
+        setState {
+            confirmationType = ConfirmationType.DELETE_CONFIRM
+            isDeleteExecutionWindowOpen = true
+            confirmLabel = ""
+            confirmMessage = "Are you sure you want to delete execution?"
+            deleteExecutionId = listOf(id)
+        }
+    }
+
+    private fun deleteExecutionBuilder(executionIds: List<Long>) {
+        val headers = Headers().also {
+            it.set("Accept", "application/json")
+            it.set("Content-Type", "application/json")
+        }
+        GlobalScope.launch {
+            responseFromDeleteExecutions =
+                    post("${window.location.origin}/deleteExecution?executionIds=${executionIds.joinToString(",")}", headers, undefined)
+        }.invokeOnCompletion {
```

Comment:
No need to use `inokeOnCompletion` here, everything can be done inside `launch`

Reply:
Fixed